src/js/pm-image-editor.js

```diff
@@ -0,0 +1,364 @@
+/*jshint multistr: true */
+(function () {
+    'use strict';
+
+    angular.module('pmImageEditor')
+        .factory('ImageEditorFactory', function () {
+            var ImageEditorFactory = function() {
+                // Visible area width. Image should always fit
+                // width or height (depends on rotation) to this area.
+                this.visibleWidth = 0;
+
+                // Set default ratio to make resetTransformations works correct.
+                this.ratio = 1;
+
+                // Real image width and height.
+                this.naturalWidth = 0;
+                this.naturalHeight = 0;
+
+                this.resetTransformations();
+            };
+
+            ImageEditorFactory.prototype.resetTransformations = function() {
+                // Coordinates of top-left corner.
+                this.top = 0;
+                this.left = 0;
+
+                // All the editor options.
+                this.selection = null;
+
+                // This variable contains rotation value by mod 2,
+                // for which latest crop was applyed.
+                this.wasCroppedForRotation = 0;
+
+                this.isCropped = false;
+                this.hFlip = false;
+                this.vFlip = false;
+
+                // Rotation value. Can be one of the following values:
+                //   0 - 0deg rotation,
+                //   1 - 90deg rotation,
+                //   2 - 180deg rotation,
+                //   3 - 270deg rotation.
+                this.rotation = 0;
```

Comment:
Why use this mapping for degrees? In the whole code we need to know this mapping otherwise code does not make much sense.

Why not using explicit degrees? instead of the mapping? 
Also on tests when you're testing what is the rotation value it's really strange rotation to return 0,1,2 or 3 .

If using degrees directly will not block anything I would suggest using degrees.

Also it will make it a lot more extensible because you can rotate for a specific value in future, like rotate just by 25deg, but with those mappings you cna only rotate by 90deg at a time.